apollo-router/src/plugins/telemetry/config_new/instruments.rs

```diff
@@ -685,6 +695,227 @@ impl InstrumentsConfig {
+                                connector_http_request_header: "content-length".to_string(),
+                                redact: None,
+                                default: None,
+                            })),
+                            selectors,
+                            updated: false,
+                        }),
+                    }
+                });
+        let http_client_response_body_size =
+            self.connector
+                .attributes
+                .http_client_response_body_size
+                .is_enabled()
+                .then(|| {
+                    let mut nb_attributes = 0;
+                    let selectors = match &self.connector.attributes.http_client_response_body_size {
+                        DefaultedStandardInstrument::Bool(_)
+                        | DefaultedStandardInstrument::Unset => None,
+                        DefaultedStandardInstrument::Extendable { attributes } => {
+                            nb_attributes = attributes.custom.len();
+                            Some(attributes.clone())
+                        }
+                    };
+                    CustomHistogram {
+                        inner: Mutex::new(CustomHistogramInner {
+                            increment: Increment::Custom(None),
+                            condition: Condition::True,
+                            histogram: Some(static_instruments
+                                .get(HTTP_CLIENT_RESPONSE_BODY_SIZE_METRIC)
+                                .expect(
+                                    "cannot get static instrument for connector; this should not happen",
+                                )
+                                .as_histogram()
+                                .cloned()
+                                .expect(
+                                    "cannot convert instrument to histogram for connector; this should not happen",
+                                )
+                            ),
+                            attributes: Vec::with_capacity(nb_attributes),
+                            selector: Some(Arc::new(ConnectorHttpSelector::ConnectorResponseHeader {
+                                connector_http_response_header: "content-length".to_string(),
+                                redact: None,
+                                default: None,
+                            })),
+                            selectors,
+                            updated: false,
+                        }),
+                    }
+                });
+        ConnectorHttpInstruments {
+            http_client_request_duration,
+            http_client_request_body_size,
+            http_client_response_body_size,
+            custom: CustomInstruments::new(&self.connector.custom, static_instruments),
+        }
+    }
+
+    pub(crate) fn new_builtin_connector_instruments(&self) -> HashMap<String, StaticInstrument> {
```

Comment:
I think this one is basically the same than subgraph instruments so probably one more reason to merge everything in subgraph

Reply:
The request and response types limit things here. You can't have attributes or selectors that are for *both* subgraph request/response *and* http request/response - they can only be generic over one type of request/response. That limits how things can be arranged.